learningServiceImpl에서 에러가 발생합니다.

스프링 부트 서버 실행 시 아래와 같은 에러가 발생하며 서버가 실행되지 않습니다.


```
Error starting ApplicationContext. To display the condition evaluation report re-run your application with 'debug' enabled.
[2m2026-04-27T11:57:35.088+09:00[0;39m [31mERROR[0;39m [35m23048[0;39m [2m--- [map-spring] [  restartedMain] [0;39m[36mo.s.boot.SpringApplication              [0;39m [2m:[0;39m Application run failed

org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'learningServiceImpl' defined in file [C:\Users\adminn\Desktop\languagemap-spring\bin\main\kr\co\mapspring\learning\service\impl\LearningServiceImpl.class]: Unsatisfied dependency expressed through constructor parameter 1: Error creating bean with name 'studyScoreRepository' defined in kr.co.mapspring.learning.repository.StudyScoreRepository defined in @EnableJpaRepositories declared on DataJpaRepositoriesRegistrar.EnableJpaRepositoriesConfiguration: No property 'id' found for type 'User'; Traversed path: StudyScore.studyLog.user
	at org.springframework.beans.factory.support.ConstructorResolver.createArgumentArray(ConstructorResolver.java:804) ~[spring-beans-7.0.6.jar:7.0.6]
	at org.springframework.beans.factory.support.ConstructorResolver.autowireConstructor(ConstructorResolver.java:240) ~[spring-beans-7.0.6.jar:7.0.6]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.autowireConstructor(AbstractAutowireCapableBeanFactory.java:1382) ~[spring-beans-7.0.6.jar:7.0.6]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1221) ~[spring-beans-7.0.6.jar:7.0.6]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:565) ~[spring-beans-7.0.6.jar:7.0.6]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:525) ~[spring-beans-7.0.6.jar:7.0.6]
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:333) ~[spring-beans-7.0.6.jar:7.0.6]
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:371) ~[spring-beans-7.0.6.jar:7.0.6]
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:331) ~[spring-beans-7.0.6.jar:7.0.6]
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:196) ~[spring-beans-7.0.6.jar:7.0.6]
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.instantiateSingleton(DefaultListableBeanFactory.java:1218) ~[spring-beans-7.0.6.jar:7.0.6]
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingleton(DefaultListableBeanFactory.java:1184) ~[spring-beans-7.0.6.jar:7.0.6]
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:1121) ~[spring-beans-7.0.6.jar:7.0.6]
	at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:994) ~[spring-context-7.0.6.jar:7.0.6]
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:621) ~[spring-context-7.0.6.jar:7.0.6]
	at org.springframework.boot.web.server.servlet.context.ServletWebServerApplicationContext.refresh(ServletWebServerApplicationContext.java:143) ~[spring-boot-web-server-4.0.5.jar:4.0.5]
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:756) ~[spring-boot-4.0.5.jar:4.0.5]
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:445) ~[spring-boot-4.0.5.jar:4.0.5]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:321) ~[spring-boot-4.0.5.jar:4.0.5]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1365) ~[spring-boot-4.0.5.jar:4.0.5]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1354) ~[spring-boot-4.0.5.jar:4.0.5]
	at kr.co.mapspring.MapSpringApplication.main(MapSpringApplication.java:10) ~[main/:na]
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103) ~[na:na]
	at java.base/java.lang.reflect.Method.invoke(Method.java:580) ~[na:na]
	at org.springframework.boot.devtools.restart.RestartLauncher.run(RestartLauncher.java:52) ~[spring-boot-devtools-4.0.5.jar:4.0.5]
Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'studyScoreRepository' defined in kr.co.mapspring.learning.repository.StudyScoreRepository defined in @EnableJpaRepositories declared on DataJpaRepositoriesRegistrar.EnableJpaRepositoriesConfiguration: No property 'id' found for type 'User'; Traversed path: StudyScore.studyLog.user
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1817) ~[spring-beans-7.0.6.jar:7.0.6]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:603) ~[spring-beans-7.0.6.jar:7.0.6]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:525) ~[spring-beans-7.0.6.jar:7.0.6]
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:333) ~[spring-beans-7.0.6.jar:7.0.6]
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:371) ~[spring-beans-7.0.6.jar:7.0.6]
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:331) ~[spring-beans-7.0.6.jar:7.0.6]
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:201) ~[spring-beans-7.0.6.jar:7.0.6]
	at org.springframework.beans.factory.config.DependencyDescriptor.resolveCandidate(DependencyDescriptor.java:229) ~[spring-beans-7.0.6.jar:7.0.6]
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.doResolveDependency(DefaultListableBeanFactory.java:1762) ~[spring-beans-7.0.6.jar:7.0.6]
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveDependency(DefaultListableBeanFactory.java:1651) ~[spring-beans-7.0.6.jar:7.0.6]
	at org.springframework.beans.factory.support.ConstructorResolver.resolveAutowiredArgument(ConstructorResolver.java:912) ~[spring-beans-7.0.6.jar:7.0.6]
	at org.springframework.beans.factory.support.ConstructorResolver.createArgumentArray(ConstructorResolver.java:791) ~[spring-beans-7.0.6.jar:7.0.6]
	... 24 common frames omitted
Caused by: org.springframework.data.repository.query.QueryCreationException: Cannot create query for method [StudyScoreRepository.findAllByStudyLog_UserId(java.lang.Long)]; No property 'id' found for type 'User'; Traversed path: StudyScore.studyLog.user
	at org.springframework.data.repository.query.QueryCreationException.create(QueryCreationException.java:109) ~[spring-data-commons-4.0.4.jar:4.0.4]
	at org.springframework.data.jpa.repository.query.PartTreeJpaQuery.<init>(PartTreeJpaQuery.java:114) ~[spring-data-jpa-4.0.4.jar:4.0.4]
	at org.springframework.data.jpa.repository.query.JpaQueryLookupStrategy$CreateQueryLookupStrategy.resolveQuery(JpaQueryLookupStrategy.java:118) ~[spring-data-jpa-4.0.4.jar:4.0.4]
	at org.springframework.data.jpa.repository.query.JpaQueryLookupStrategy$CreateIfNotFoundQueryLookupStrategy.resolveQuery(JpaQueryLookupStrategy.java:291) ~[spring-data-jpa-4.0.4.jar:4.0.4]
	at org.springframework.data.jpa.repository.query.JpaQueryLookupStrategy$AbstractQueryLookupStrategy.resolveQuery(JpaQueryLookupStrategy.java:93) ~[spring-data-jpa-4.0.4.jar:4.0.4]
	at org.springframework.data.repository.core.support.QueryExecutorMethodInterceptor.lookupQuery(QueryExecutorMethodInterceptor.java:114) ~[spring-data-commons-4.0.4.jar:4.0.4]
	at org.springframework.data.repository.core.support.QueryExecutorMethodInterceptor.mapMethodsToQuery(QueryExecutorMethodInterceptor.java:102) ~[spring-data-commons-4.0.4.jar:4.0.4]
	at org.springframework.data.repository.core.support.QueryExecutorMethodInterceptor.<init>(QueryExecutorMethodInterceptor.java:90) ~[spring-data-commons-4.0.4.jar:4.0.4]
	at org.springframework.data.repository.core.support.RepositoryFactorySupport.getRepository(RepositoryFactorySupport.java:413) ~[spring-data-commons-4.0.4.jar:4.0.4]
	at org.springframework.data.repository.core.support.RepositoryFactoryBeanSupport.lambda$afterPropertiesSet$1(RepositoryFactoryBeanSupport.java:361) ~[spring-data-commons-4.0.4.jar:4.0.4]
	at org.springframework.data.util.Lazy.getNullable(Lazy.java:136) ~[spring-data-commons-4.0.4.jar:4.0.4]
	at org.springframework.data.util.Lazy.get(Lazy.java:114) ~[spring-data-commons-4.0.4.jar:4.0.4]
	at org.springframework.data.repository.core.support.RepositoryFactoryBeanSupport.afterPropertiesSet(RepositoryFactoryBeanSupport.java:370) ~[spring-data-commons-4.0.4.jar:4.0.4]
	at org.springframework.data.jpa.repository.support.JpaRepositoryFactoryBean.afterPropertiesSet(JpaRepositoryFactoryBean.java:212) ~[spring-data-jpa-4.0.4.jar:4.0.4]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.invokeInitMethods(AbstractAutowireCapableBeanFactory.java:1864) ~[spring-beans-7.0.6.jar:7.0.6]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1813) ~[spring-beans-7.0.6.jar:7.0.6]
	... 35 common frames omitted
Caused by: org.springframework.data.core.PropertyReferenceException: No property 'id' found for type 'User'; Traversed path: StudyScore.studyLog.user
	at org.springframework.data.core.SimplePropertyPath.<init>(SimplePropertyPath.java:93) ~[spring-data-commons-4.0.4.jar:4.0.4]
	at org.springframework.data.core.SimplePropertyPath.create(SimplePropertyPath.java:360) ~[spring-data-commons-4.0.4.jar:4.0.4]
	at org.springframework.data.core.SimplePropertyPath.create(SimplePropertyPath.java:335) ~[spring-data-commons-4.0.4.jar:4.0.4]
	at org.springframework.data.core.SimplePropertyPath.create(SimplePropertyPath.java:370) ~[spring-data-commons-4.0.4.jar:4.0.4]
	at org.springframework.data.core.SimplePropertyPath.create(SimplePropertyPath.java:393) ~[spring-data-commons-4.0.4.jar:4.0.4]
	at org.springframework.data.core.SimplePropertyPath.create(SimplePropertyPath.java:335) ~[spring-data-commons-4.0.4.jar:4.0.4]
	at org.springframework.data.core.SimplePropertyPath.create(SimplePropertyPath.java:319) ~[spring-data-commons-4.0.4.jar:4.0.4]
	at org.springframework.data.core.SimplePropertyPath.lambda$from$0(SimplePropertyPath.java:291) ~[spring-data-commons-4.0.4.jar:4.0.4]
	at org.springframework.util.ConcurrentReferenceHashMap$6.execute(ConcurrentReferenceHashMap.java:387) ~[spring-core-7.0.6.jar:7.0.6]
	at org.springframework.util.ConcurrentReferenceHashMap$Segment.doTask(ConcurrentReferenceHashMap.java:681) ~[spring-core-7.0.6.jar:7.0.6]
	at org.springframework.util.ConcurrentReferenceHashMap.doTask(ConcurrentReferenceHashMap.java:565) ~[spring-core-7.0.6.jar:7.0.6]
	at org.springframework.util.ConcurrentReferenceHashMap.computeIfAbsent(ConcurrentReferenceHashMap.java:381) ~[spring-core-7.0.6.jar:7.0.6]
	at org.springframework.data.core.SimplePropertyPath.from(SimplePropertyPath.java:270) ~[spring-data-commons-4.0.4.jar:4.0.4]
	at org.springframework.data.core.PropertyPath.from(PropertyPath.java:210) ~[spring-data-commons-4.0.4.jar:4.0.4]
	at org.springframework.data.core.PropertyPath.from(PropertyPath.java:193) ~[spring-data-commons-4.0.4.jar:4.0.4]
	at org.springframework.data.repository.query.parser.Part.<init>(Part.java:81) ~[spring-data-commons-4.0.4.jar:4.0.4]
	at org.springframework.data.repository.query.parser.PartTree$OrPart.lambda$new$0(PartTree.java:260) ~[spring-data-commons-4.0.4.jar:4.0.4]
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197) ~[na:na]
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:179) ~[na:na]
	at java.base/java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:1024) ~[na:na]
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509) ~[na:na]
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499) ~[na:na]
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921) ~[na:na]
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) ~[na:na]
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:682) ~[na:na]
	at org.springframework.data.repository.query.parser.PartTree$OrPart.<init>(PartTree.java:261) ~[spring-data-commons-4.0.4.jar:4.0.4]
	at org.springframework.data.repository.query.parser.PartTree$Predicate.lambda$new$0(PartTree.java:390) ~[spring-data-commons-4.0.4.jar:4.0.4]
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197) ~[na:na]
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:179) ~[na:na]
	at java.base/java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:1024) ~[na:na]
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509) ~[na:na]
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499) ~[na:na]
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921) ~[na:na]
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) ~[na:na]
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:682) ~[na:na]
	at org.springframework.data.repository.query.parser.PartTree$Predicate.<init>(PartTree.java:391) ~[spring-data-commons-4.0.4.jar:4.0.4]
	at org.springframework.data.repository.query.parser.PartTree.<init>(PartTree.java:104) ~[spring-data-commons-4.0.4.jar:4.0.4]
	at org.springframework.data.jpa.repository.query.PartTreeJpaQuery.<init>(PartTreeJpaQuery.java:108) ~[spring-data-jpa-4.0.4.jar:4.0.4]
	... 49 common frames omitted

```

--- a/src/main/java/kr/co/mapspring/global/exception/GlobalExceptionHandler.java
+++ b/src/main/java/kr/co/mapspring/global/exception/GlobalExceptionHandler.java
@@ -3,6 +3,10 @@ package kr.co.mapspring.global.exception;
 import java.util.HashMap;
 import java.util.Map;
 
+import kr.co.mapspring.global.exception.learning.GoalAlreadySelectedException;
+import kr.co.mapspring.global.exception.learning.GoalMasterNotFoundException;
+import kr.co.mapspring.global.exception.learning.GoalSelectionLimitExceededException;
+import kr.co.mapspring.global.exception.learning.UserGoalNotFoundException;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
 import org.springframework.http.converter.HttpMessageConversionException;
@@ -67,6 +71,41 @@ public class GlobalExceptionHandler {
         return ResponseEntity.status(HttpStatus.BAD_REQUEST)
                 .body(ApiResponseDTO.fail(
                         HttpStatus.BAD_REQUEST,
+                        e.getMessage()
+                ));
+    }
+
+    // Learning 목표 관련 예외 처리
+    @ExceptionHandler({
+            GoalSelectionLimitExceededException.class
+    })
+    public ResponseEntity<ApiResponseDTO<Object>> handleLearningBadRequestException(RuntimeException e) {
+        return ResponseEntity.status(HttpStatus.BAD_REQUEST)
+                .body(ApiResponseDTO.fail(
+                        HttpStatus.BAD_REQUEST,
+                        e.getMessage()
+                ));
+    }
+
+    @ExceptionHandler({
+            GoalMasterNotFoundException.class,
+            UserGoalNotFoundException.class
+    })
+    public ResponseEntity<ApiResponseDTO<Object>> handleLearningNotFoundException(RuntimeException e) {
+        return ResponseEntity.status(HttpStatus.NOT_FOUND)
+                .body(ApiResponseDTO.fail(
+                        HttpStatus.NOT_FOUND,
+                        e.getMessage()
+                ));
+    }
+
+    @ExceptionHandler(GoalAlreadySelectedException.class)
+    public ResponseEntity<ApiResponseDTO<Object>> handleGoalAlreadySelectedException(
+            GoalAlreadySelectedException e
+    ) {
+        return ResponseEntity.status(HttpStatus.CONFLICT)
+                .body(ApiResponseDTO.fail(
+                        HttpStatus.CONFLICT,
                         e.getMessage()
                 ));
     }

--- a/src/main/java/kr/co/mapspring/global/exception/learning/GoalAlreadySelectedException.java
+++ b/src/main/java/kr/co/mapspring/global/exception/learning/GoalAlreadySelectedException.java
@@ -1,7 +1,10 @@
 package kr.co.mapspring.global.exception.learning;
 
-public class GoalAlreadySelectedException extends RuntimeException {
+import kr.co.mapspring.global.exception.CustomException;
+import kr.co.mapspring.global.exception.ErrorCode;
 
-    public GoalAlreadySelectedException() {super("이미 선택한 학습 목표입니다.");}
+public class GoalAlreadySelectedException extends CustomException {
+
+    public GoalAlreadySelectedException() {super(ErrorCode.GOAL_ALREADY_SELECTED);}
 
 }

--- a/src/main/java/kr/co/mapspring/global/exception/learning/GoalMasterNotFoundException.java
+++ b/src/main/java/kr/co/mapspring/global/exception/learning/GoalMasterNotFoundException.java
@@ -1,7 +1,10 @@
 package kr.co.mapspring.global.exception.learning;
 
-public class GoalMasterNotFoundException extends RuntimeException {
+import kr.co.mapspring.global.exception.CustomException;
+import kr.co.mapspring.global.exception.ErrorCode;
 
-    public GoalMasterNotFoundException() { super("존재하지 않는 학습 목표입니다."); }
+public class GoalMasterNotFoundException extends CustomException {
+
+    public GoalMasterNotFoundException() { super(ErrorCode.GOAL_MASTER_NOT_FOUND); }
 
 }

--- a/src/main/java/kr/co/mapspring/global/exception/learning/GoalSelectionLimitExceededException.java
+++ b/src/main/java/kr/co/mapspring/global/exception/learning/GoalSelectionLimitExceededException.java
@@ -1,7 +1,10 @@
 package kr.co.mapspring.global.exception.learning;
 
-public class GoalSelectionLimitExceededException extends RuntimeException {
+import kr.co.mapspring.global.exception.CustomException;
+import kr.co.mapspring.global.exception.ErrorCode;
 
-    public GoalSelectionLimitExceededException() { super("학습 목표는 최대 3개까지만 선택할 수 있습니다."); }
+public class GoalSelectionLimitExceededException extends CustomException {
+
+    public GoalSelectionLimitExceededException() { super(ErrorCode.GOAL_SELECTION_LIMIT_EXCEEDED); }
 
 }

--- a/src/main/java/kr/co/mapspring/global/exception/learning/UserGoalNotFoundException.java
+++ b/src/main/java/kr/co/mapspring/global/exception/learning/UserGoalNotFoundException.java
@@ -1,7 +1,10 @@
 package kr.co.mapspring.global.exception.learning;
 
-public class UserGoalNotFoundException extends RuntimeException {
+import kr.co.mapspring.global.exception.CustomException;
+import kr.co.mapspring.global.exception.ErrorCode;
 
-    public UserGoalNotFoundException() { super("존재하지 않는 사용자 목표입니다."); }
+public class UserGoalNotFoundException extends CustomException {
+
+    public UserGoalNotFoundException() { super(ErrorCode.USER_GOAL_NOT_FOUND); }
 
 }

--- a/src/main/java/kr/co/mapspring/learning/controller/LearningGoalController.java
+++ b/src/main/java/kr/co/mapspring/learning/controller/LearningGoalController.java
@@ -51,7 +51,7 @@ public class LearningGoalController implements LearningGoalControllerDocs {
     }
 
     @Override
-    @GetMapping("completed")
+    @GetMapping("/completed")
     public ResponseEntity<ApiResponseDTO<List<LearningGoalDto.ResponseUserGoal>>> getCompletedGoals(@RequestParam("userId") Long userId) {
         List<UserGoal> goals = learningGoalService.getCompletedGoals(userId);
 

--- a/src/main/java/kr/co/mapspring/learning/controller/docs/LearningControllerDocs.java
+++ b/src/main/java/kr/co/mapspring/learning/controller/docs/LearningControllerDocs.java
@@ -1,44 +1,108 @@
 package kr.co.mapspring.learning.controller.docs;
 
 import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.Parameter;
 import io.swagger.v3.oas.annotations.media.Content;
+import io.swagger.v3.oas.annotations.media.ExampleObject;
 import io.swagger.v3.oas.annotations.media.Schema;
 import io.swagger.v3.oas.annotations.responses.ApiResponse;
 import io.swagger.v3.oas.annotations.responses.ApiResponses;
+import io.swagger.v3.oas.annotations.tags.Tag;
 import kr.co.mapspring.global.dto.ApiResponseDTO;
 import kr.co.mapspring.learning.dto.LearningLogDto;
 import org.springframework.http.ResponseEntity;
 
 import java.util.List;
 
+@Tag(name = "Learning", description = "학습 관련 API")
 public interface LearningControllerDocs {
 
     @Operation(
             summary = "학습 기록 조회",
             description = "사용자의 학습 기록과 점수 정보를 조회합니다."
     )
-    @ApiResponses(value = {
-            @ApiResponse(responseCode = "200",
+    @ApiResponses({
+            @ApiResponse(
+                    responseCode = "200",
                     description = "학습 기록 조회 성공",
                     content = @Content(
                             mediaType = "application/json",
-                            schema = @Schema(implementation = ApiResponseDTO.class)
+                            schema = @Schema(implementation = ApiResponseDTO.class),
+                            examples = @ExampleObject(
+                                    value = """
+                                            {
+                                              "success": true,
+                                              "status": 200,
+                                              "message": "학습 기록 조회 성공",
+                                              "data": [
+                                                {
+                                                  "studyLogId": 1,
+                                                  "studyType": "SCENARIO",
+                                                  "earnedExp": 20,
+                                                  "naturalnessScore": 80,
+                                                  "fluencyScore": 70,
+                                                  "totalScore": 75
+                                                }
+                                              ]
+                                            }
+                                            """
+                            )
                     )
             ),
-            @ApiResponse(responseCode = "400",
+            @ApiResponse(
+                    responseCode = "400",
                     description = "잘못된 요청값",
                     content = @Content(
                             mediaType = "application/json",
-                            schema = @Schema(implementation = ApiResponseDTO.class)
+                            examples = @ExampleObject(
+                                    value = """
+                                            {
+                                              "success": false,
+                                              "status": 400,
+                                              "message": "잘못된 요청입니다.",
+                                              "data": null
+                                            }
+                                            """
+                            )
                     )
             ),
-            @ApiResponse(responseCode = "500",
+            @ApiResponse(
+                    responseCode = "404",
+                    description = "사용자를 찾을 수 없음",
+                    content = @Content(
+                            mediaType = "application/json",
+                            examples = @ExampleObject(
+                                    value = """
+                                            {
+                                              "success": false,
+                                              "status": 404,
+                                              "message": "사용자를 찾을 수 없습니다.",
+                                              "data": null
+                                            }
+                                            """
+                            )
+                    )
+            ),
+            @ApiResponse(
+                    responseCode = "500",
                     description = "서버 내부 오류",
                     content = @Content(
                             mediaType = "application/json",
-                            schema = @Schema(implementation = ApiResponseDTO.class)
+                            examples = @ExampleObject(
+                                    value = """
+                                            {
+                                              "success": false,
+                                              "status": 500,
+                                              "message": "서버 내부 오류입니다.",
+                                              "data": null
+                                            }
+                                            """
+                            )
                     )
             )
     })
-    ResponseEntity<ApiResponseDTO<List<LearningLogDto.ResponseLog>>> getStudyLogs(Long userId);
+    ResponseEntity<ApiResponseDTO<List<LearningLogDto.ResponseLog>>> getStudyLogs(
+            @Parameter(description = "사용자 ID", example = "1")
+            Long userId
+    );
 }

--- a/src/main/java/kr/co/mapspring/learning/controller/docs/LearningGoalControllerDocs.java
+++ b/src/main/java/kr/co/mapspring/learning/controller/docs/LearningGoalControllerDocs.java
@@ -1,91 +1,280 @@
 package kr.co.mapspring.learning.controller.docs;
 
 import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.Parameter;
 import io.swagger.v3.oas.annotations.media.Content;
+import io.swagger.v3.oas.annotations.media.ExampleObject;
 import io.swagger.v3.oas.annotations.media.Schema;
+import io.swagger.v3.oas.annotations.parameters.RequestBody;
 import io.swagger.v3.oas.annotations.responses.ApiResponse;
 import io.swagger.v3.oas.annotations.responses.ApiResponses;
+import io.swagger.v3.oas.annotations.tags.Tag;
 import kr.co.mapspring.global.dto.ApiResponseDTO;
 import kr.co.mapspring.learning.dto.LearningGoalDto;
 import org.springframework.http.ResponseEntity;
 
 import java.util.List;
 
+@Tag(name = "Learning Goal", description = "학습 목표 관련 API")
 public interface LearningGoalControllerDocs {
 
     @Operation(
             summary = "학습 목표 선택",
-            description = "사용자가 학습 목표를 선택합니다."
+            description = "사용자가 학습 목표를 선택합니다. 사용자는 최대 3개의 목표까지만 선택할 수 있습니다."
     )
-    @ApiResponses(value = {
-            @ApiResponse(responseCode = "200",
+    @ApiResponses({
+            @ApiResponse(
+                    responseCode = "200",
                     description = "목표 선택 성공",
-                    content = @Content(schema = @Schema(implementation = ApiResponseDTO.class))
+                    content = @Content(
+                            mediaType = "application/json",
+                            examples = @ExampleObject(
+                                    value = """
+                                            {
+                                              "success": true,
+                                              "status": 200,
+                                              "message": "목표 선택 성공",
+                                              "data": null
+                                            }
+                                            """
+                            )
+                    )
             ),
-            @ApiResponse(responseCode = "400",
+            @ApiResponse(
+                    responseCode = "400",
                     description = "잘못된 요청 또는 목표 선택 제한 초과",
-                    content = @Content(schema = @Schema(implementation = ApiResponseDTO.class))
+                    content = @Content(
+                            mediaType = "application/json",
+                            examples = @ExampleObject(
+                                    value = """
+                                            {
+                                              "success": false,
+                                              "status": 400,
+                                              "message": "학습 목표는 최대 3개까지만 선택할 수 있습니다.",
+                                              "data": null
+                                            }
+                                            """
+                            )
+                    )
             ),
-            @ApiResponse(responseCode = "409",
+            @ApiResponse(
+                    responseCode = "404",
+                    description = "목표 마스터를 찾을 수 없음",
+                    content = @Content(
+                            mediaType = "application/json",
+                            examples = @ExampleObject(
+                                    value = """
+                                            {
+                                              "success": false,
+                                              "status": 404,
+                                              "message": "존재하지 않는 학습 목표입니다.",
+                                              "data": null
+                                            }
+                                            """
+                            )
+                    )
+            ),
+            @ApiResponse(
+                    responseCode = "409",
                     description = "이미 선택한 목표",
-                    content = @Content(schema = @Schema(implementation = ApiResponseDTO.class))
+                    content = @Content(
+                            mediaType = "application/json",
+                            examples = @ExampleObject(
+                                    value = """
+                                            {
+                                              "success": false,
+                                              "status": 409,
+                                              "message": "이미 선택한 학습 목표입니다.",
+                                              "data": null
+                                            }
+                                            """
+                            )
+                    )
             )
     })
-    ResponseEntity<ApiResponseDTO<Void>> selectGoal(LearningGoalDto.RequestSelectGoal request);
-
+    ResponseEntity<ApiResponseDTO<Void>> selectGoal(
+            @RequestBody(
+                    description = "학습 목표 선택 요청 정보",
+                    required = true,
+                    content = @Content(
+                            mediaType = "application/json",
+                            schema = @Schema(implementation = LearningGoalDto.RequestSelectGoal.class),
+                            examples = @ExampleObject(
+                                    value = """
+                                            {
+                                              "userId": 1,
+                                              "goalMasterId": 10
+                                            }
+                                            """
+                            )
+                    )
+            )
+            LearningGoalDto.RequestSelectGoal request
+    );
 
     @Operation(
             summary = "학습 목표 해제",
             description = "사용자가 선택한 학습 목표를 해제합니다."
     )
-    @ApiResponses(value = {
-            @ApiResponse(responseCode = "200",
+    @ApiResponses({
+            @ApiResponse(
+                    responseCode = "200",
                     description = "목표 해제 성공",
-                    content = @Content(schema = @Schema(implementation = ApiResponseDTO.class))
+                    content = @Content(
+                            mediaType = "application/json",
+                            examples = @ExampleObject(
+                                    value = """
+                                            {
+                                              "success": true,
+                                              "status": 200,
+                                              "message": "목표 해제 성공",
+                                              "data": null
+                                            }
+                                            """
+                            )
+                    )
             ),
-            @ApiResponse(responseCode = "404",
-                    description = "해당 목표를 찾을 수 없음",
-                    content = @Content(schema = @Schema(implementation = ApiResponseDTO.class))
+            @ApiResponse(
+                    responseCode = "404",
+                    description = "해당 사용자 목표를 찾을 수 없음",
+                    content = @Content(
+                            mediaType = "application/json",
+                            examples = @ExampleObject(
+                                    value = """
+                                            {
+                                              "success": false,
+                                              "status": 404,
+                                              "message": "사용자 목표를 찾을 수 없습니다.",
+                                              "data": null
+                                            }
+                                            """
+                            )
+                    )
             )
     })
-    ResponseEntity<ApiResponseDTO<Void>> cancelGoal(Long userGoalId);
-
+    ResponseEntity<ApiResponseDTO<Void>> cancelGoal(
+            @Parameter(description = "사용자 목표 ID", example = "1")
+            Long userGoalId
+    );
 
     @Operation(
             summary = "진행 중 학습 목표 조회",
             description = "사용자의 진행 중인 학습 목표 목록을 조회합니다."
     )
-    @ApiResponses(value = {
-            @ApiResponse(responseCode = "200",
-                    description = "조회 성공",
-                    content = @Content(schema = @Schema(implementation = ApiResponseDTO.class))
+    @ApiResponses({
+            @ApiResponse(
+                    responseCode = "200",
+                    description = "진행 중 목표 조회 성공",
+                    content = @Content(
+                            mediaType = "application/json",
+                            schema = @Schema(implementation = ApiResponseDTO.class),
+                            examples = @ExampleObject(
+                                    value = """
+                                            {
+                                              "success": true,
+                                              "status": 200,
+                                              "message": "진행 중 학습 목표 조회 성공",
+                                              "data": [
+                                                {
+                                                  "userGoalId": 1,
+                                                  "goalMasterId": 10,
+                                                  "goalTitle": "하루 1회 학습",
+                                                  "goalType": "STUDY_COUNT",
+                                                  "periodType": "DAILY",
+                                                  "targetValue": 1,
+                                                  "currentValue": 0,
+                                                  "status": "ACTIVE",
+                                                  "startDate": "2026-04-27",
+                                                  "endDate": "2026-04-27"
+                                                }
+                                              ]
+                                            }
+                                            """
+                            )
+                    )
             )
     })
-    ResponseEntity<ApiResponseDTO<List<LearningGoalDto.ResponseUserGoal>>> getActiveGoals(Long userId);
-
+    ResponseEntity<ApiResponseDTO<List<LearningGoalDto.ResponseUserGoal>>> getActiveGoals(
+            @Parameter(description = "사용자 ID", example = "1")
+            Long userId
+    );
 
     @Operation(
             summary = "완료된 학습 목표 조회",
             description = "사용자의 완료된 학습 목표 목록을 조회합니다."
     )
-    @ApiResponses(value = {
-            @ApiResponse(responseCode = "200",
-                    description = "조회 성공",
-                    content = @Content(schema = @Schema(implementation = ApiResponseDTO.class))
+    @ApiResponses({
+            @ApiResponse(
+                    responseCode = "200",
+                    description = "완료된 목표 조회 성공",
+                    content = @Content(
+                            mediaType = "application/json",
+                            schema = @Schema(implementation = ApiResponseDTO.class),
+                            examples = @ExampleObject(
+                                    value = """
+                                            {
+                                              "success": true,
+                                              "status": 200,
+                                              "message": "완료된 학습 목표 조회 성공",
+                                              "data": [
+                                                {
+                                                  "userGoalId": 2,
+                                                  "goalMasterId": 10,
+                                                  "goalTitle": "하루 1회 학습",
+                                                  "goalType": "STUDY_COUNT",
+                                                  "periodType": "DAILY",
+                                                  "targetValue": 1,
+                                                  "currentValue": 1,
+                                                  "status": "COMPLETED",
+                                                  "startDate": "2026-04-27",
+                                                  "endDate": "2026-04-27"
+                                                }
+                                              ]
+                                            }
+                                            """
+                            )
+                    )
             )
     })
-    ResponseEntity<ApiResponseDTO<List<LearningGoalDto.ResponseUserGoal>>> getCompletedGoals(Long userId);
-
+    ResponseEntity<ApiResponseDTO<List<LearningGoalDto.ResponseUserGoal>>> getCompletedGoals(
+            @Parameter(description = "사용자 ID", example = "1")
+            Long userId
+    );
 
     @Operation(
             summary = "선택 가능한 학습 목표 조회",
-            description = "사용자가 선택할 수 있는 학습 목표 목록을 조회합니다."
+            description = "사용자가 아직 선택하지 않은 학습 목표 목록을 조회합니다."
     )
-    @ApiResponses(value = {
-            @ApiResponse(responseCode = "200",
-                    description = "조회 성공",
-                    content = @Content(schema = @Schema(implementation = ApiResponseDTO.class))
+    @ApiResponses({
+            @ApiResponse(
+                    responseCode = "200",
+                    description = "선택 가능한 목표 조회 성공",
+                    content = @Content(
+                            mediaType = "application/json",
+                            schema = @Schema(implementation = ApiResponseDTO.class),
+                            examples = @ExampleObject(
+                                    value = """
+                                            {
+                                              "success": true,
+                                              "status": 200,
+                                              "message": "선택 가능한 학습 목표 조회 성공",
+                                              "data": [
+                                                {
+                                                  "goalMasterId": 11,
+                                                  "goalTitle": "하루 3회 학습",
+                                                  "goalType": "STUDY_COUNT",
+                                                  "periodType": "DAILY",
+                                                  "targetValue": 3
+                                                }
+                                              ]
+                                            }
+                                            """
+                            )
+                    )
             )
     })
-    ResponseEntity<ApiResponseDTO<List<LearningGoalDto.ResponseGoalMaster>>> getAvailableGoals(Long userId);
+    ResponseEntity<ApiResponseDTO<List<LearningGoalDto.ResponseGoalMaster>>> getAvailableGoals(
+            @Parameter(description = "사용자 ID", example = "1")
+            Long userId
+    );
 }

--- a/src/main/java/kr/co/mapspring/learning/repository/StudyScoreRepository.java
+++ b/src/main/java/kr/co/mapspring/learning/repository/StudyScoreRepository.java
@@ -7,5 +7,5 @@ import java.util.List;
 
 public interface StudyScoreRepository extends JpaRepository<StudyScore, Long> {
 
-    List<StudyScore> findAllByStudyLog_UserId(Long userId);
+    List<StudyScore> findAllByStudyLog_User_UserId(Long userId);
 }

--- a/src/main/java/kr/co/mapspring/learning/repository/UserGoalRepository.java
+++ b/src/main/java/kr/co/mapspring/learning/repository/UserGoalRepository.java
@@ -8,9 +8,9 @@ import java.util.List;
 
 public interface UserGoalRepository extends JpaRepository<UserGoal, Long> {
 
-    boolean existsByUser_UserIdAndGoalMaster_GoalMasterId(Long userId, Long goalMasterId);
+    boolean existsByUser_UserIdAndGoalMaster_GoalMasterIdAndStatus(Long userId, Long goalMasterId, UserGoalStatus status);
 
-    int countByUser_UserId(Long userId);
+    int countByUser_UserIdAndStatus(Long userId, UserGoalStatus status);
 
     List<UserGoal> findAllByUser_UserIdAndStatus(Long userId, UserGoalStatus status);
 

--- a/src/main/java/kr/co/mapspring/learning/service/impl/LearningGoalServiceImpl.java
+++ b/src/main/java/kr/co/mapspring/learning/service/impl/LearningGoalServiceImpl.java
@@ -51,7 +51,7 @@ public class LearningGoalServiceImpl implements LearningGoalService {
 
         int selectedCount = userGoalRepository.countByUser_UserId(userId);
 
-        if (selectedCount > MAX_GOAL_COUNT) {
+        if (selectedCount >= MAX_GOAL_COUNT) {
             throw new GoalSelectionLimitExceededException();
         }
 

--- a/src/main/java/kr/co/mapspring/learning/service/impl/LearningGoalServiceImpl.java
+++ b/src/main/java/kr/co/mapspring/learning/service/impl/LearningGoalServiceImpl.java
@@ -43,13 +43,13 @@ public class LearningGoalServiceImpl implements LearningGoalService {
                 .orElseThrow(GoalMasterNotFoundException::new);
 
         boolean alreadySelected =
-                userGoalRepository.existsByUser_UserIdAndGoalMaster_GoalMasterId(userId, goalMasterId);
+                userGoalRepository.existsByUser_UserIdAndGoalMaster_GoalMasterIdAndStatus(userId, goalMasterId, UserGoalStatus.ACTIVE);
 
         if (alreadySelected) {
             throw new GoalAlreadySelectedException();
         }
 
-        int selectedCount = userGoalRepository.countByUser_UserId(userId);
+        int selectedCount = userGoalRepository.countByUser_UserIdAndStatus(userId, UserGoalStatus.ACTIVE);
 
         if (selectedCount >= MAX_GOAL_COUNT) {
             throw new GoalSelectionLimitExceededException();

--- a/src/main/java/kr/co/mapspring/learning/service/impl/LearningServiceImpl.java
+++ b/src/main/java/kr/co/mapspring/learning/service/impl/LearningServiceImpl.java
@@ -65,7 +65,7 @@ public class LearningServiceImpl implements LearningService {
 
     @Override
     public List<LearningLogDto.ResponseLog> getStudyLogs(Long userId) {
-        List<StudyScore> studyScores = studyScoreRepository.findAllByStudyLog_UserId(userId);
+        List<StudyScore> studyScores = studyScoreRepository.findAllByStudyLog_User_UserId(userId);
 
         return studyScores.stream()
                 .map(studyScore -> LearningLogDto.ResponseLog.from(

--- a/src/test/java/kr/co/mapspring/learning/service/LearningGoalServiceTest.java
+++ b/src/test/java/kr/co/mapspring/learning/service/LearningGoalServiceTest.java
@@ -12,6 +12,10 @@ import kr.co.mapspring.learning.enums.UserGoalStatus;
 import kr.co.mapspring.learning.repository.GoalMasterRepository;
 import kr.co.mapspring.learning.repository.UserGoalRepository;
 import kr.co.mapspring.learning.service.impl.LearningGoalServiceImpl;
+import kr.co.mapspring.user.entity.User;
+import kr.co.mapspring.user.enums.UserRole;
+import kr.co.mapspring.user.enums.UserStatus;
+import kr.co.mapspring.user.repository.UserRepository;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.DisplayName;
 import org.junit.jupiter.api.Test;
@@ -42,17 +46,34 @@ class LearningGoalServiceTest {
     @Mock
     private UserGoalRepository userGoalRepository;
 
+    @Mock
+    private UserRepository userRepository;
+
     @InjectMocks
     private LearningGoalServiceImpl learningGoalService;
 
     private Long userId;
     private Long goalMasterId;
+    private User user;
     private GoalMaster goalMaster;
 
     @BeforeEach
     void setUp() {
         userId = 1L;
         goalMasterId = 100L;
+
+        user = User.builder()
+                .userId(userId)
+                .email("test@test.com")
+                .name("테스트")
+                .birthDate(LocalDate.of(2000, 1, 1))
+                .address("서울")
+                .phoneNumber("010-1234-5678")
+                .passwordHash("1234")
+                .status(UserStatus.ACTIVE)
+                .role(UserRole.USER)
+                .build();
+
         goalMaster = GoalMaster.of(
                 goalMasterId,
                 "하루 1회 학습",
@@ -65,30 +86,31 @@ class LearningGoalServiceTest {
     @Test
     @DisplayName("학습 목표를 정상적으로 선택한다")
     void 학습_목표를_정상적으로_선택한다() {
-        // given
+        given(userRepository.findById(userId))
+                .willReturn(Optional.of(user));
+
         given(goalMasterRepository.findById(goalMasterId))
                 .willReturn(Optional.of(goalMaster));
 
-        given(userGoalRepository.existsByUserIdAndGoalMaster_GoalMasterId(userId, goalMasterId))
+        given(userGoalRepository.existsByUser_UserIdAndGoalMaster_GoalMasterId(userId, goalMasterId))
                 .willReturn(false);
 
-        given(userGoalRepository.countByUserId(userId))
+        given(userGoalRepository.countByUser_UserId(userId))
                 .willReturn(2);
 
-        // when
         learningGoalService.selectGoal(userId, goalMasterId);
 
-        // then
+        verify(userRepository).findById(userId);
         verify(goalMasterRepository).findById(goalMasterId);
-        verify(userGoalRepository).existsByUserIdAndGoalMaster_GoalMasterId(userId, goalMasterId);
-        verify(userGoalRepository).countByUserId(userId);
+        verify(userGoalRepository).existsByUser_UserIdAndGoalMaster_GoalMasterId(userId, goalMasterId);
+        verify(userGoalRepository).countByUser_UserId(userId);
 
         ArgumentCaptor<UserGoal> captor = ArgumentCaptor.forClass(UserGoal.class);
         verify(userGoalRepository).save(captor.capture());
 
         UserGoal savedUserGoal = captor.getValue();
 
-        assertEquals(userId, savedUserGoal.getUserId());
+        assertEquals(userId, savedUserGoal.getUser().getUserId());
         assertEquals(goalMasterId, savedUserGoal.getGoalMaster().getGoalMasterId());
         assertEquals(0, savedUserGoal.getCurrentValue());
         assertEquals(UserGoalStatus.ACTIVE, savedUserGoal.getStatus());
@@ -99,85 +121,89 @@ class LearningGoalServiceTest {
     @Test
     @DisplayName("존재하지 않는 목표는 선택할 수 없다")
     void 존재하지_않는_목표는_선택할_수_없다() {
-        // given
+        given(userRepository.findById(userId))
+                .willReturn(Optional.of(user));
+
         given(goalMasterRepository.findById(goalMasterId))
                 .willReturn(Optional.empty());
 
-        // when & then
         assertThrows(GoalMasterNotFoundException.class,
                 () -> learningGoalService.selectGoal(userId, goalMasterId));
 
+        verify(userRepository).findById(userId);
         verify(goalMasterRepository).findById(goalMasterId);
-        verify(userGoalRepository, never()).existsByUserIdAndGoalMaster_GoalMasterId(any(), any());
-        verify(userGoalRepository, never()).countByUserId(any());
+        verify(userGoalRepository, never()).existsByUser_UserIdAndGoalMaster_GoalMasterId(any(), any());
+        verify(userGoalRepository, never()).countByUser_UserId(any());
         verify(userGoalRepository, never()).save(any(UserGoal.class));
     }
 
     @Test
     @DisplayName("이미 선택한 목표는 중복 선택할 수 없다")
     void 이미_선택한_목표는_중복_선택할_수_없다() {
-        // given
+        given(userRepository.findById(userId))
+                .willReturn(Optional.of(user));
+
         given(goalMasterRepository.findById(goalMasterId))
                 .willReturn(Optional.of(goalMaster));
 
-        given(userGoalRepository.existsByUserIdAndGoalMaster_GoalMasterId(userId, goalMasterId))
+        given(userGoalRepository.existsByUser_UserIdAndGoalMaster_GoalMasterId(userId, goalMasterId))
                 .willReturn(true);
 
-        // when & then
         assertThrows(GoalAlreadySelectedException.class,
                 () -> learningGoalService.selectGoal(userId, goalMasterId));
 
+        verify(userRepository).findById(userId);
         verify(goalMasterRepository).findById(goalMasterId);
-        verify(userGoalRepository).existsByUserIdAndGoalMaster_GoalMasterId(userId, goalMasterId);
-        verify(userGoalRepository, never()).countByUserId(any());
+        verify(userGoalRepository).existsByUser_UserIdAndGoalMaster_GoalMasterId(userId, goalMasterId);
+        verify(userGoalRepository, never()).countByUser_UserId(any());
         verify(userGoalRepository, never()).save(any(UserGoal.class));
     }
 
     @Test
     @DisplayName("학습 목표는 최대 3개까지만 선택할 수 있다")
     void 학습_목표는_최대_3개까지만_선택할_수_있다() {
-        // given
+        given(userRepository.findById(userId))
+                .willReturn(Optional.of(user));
+
         given(goalMasterRepository.findById(goalMasterId))
                 .willReturn(Optional.of(goalMaster));
 
-        given(userGoalRepository.existsByUserIdAndGoalMaster_GoalMasterId(userId, goalMasterId))
+        given(userGoalRepository.existsByUser_UserIdAndGoalMaster_GoalMasterId(userId, goalMasterId))
                 .willReturn(false);
 
-        given(userGoalRepository.countByUserId(userId))
-                .willReturn(4);
+        given(userGoalRepository.countByUser_UserId(userId))
+                .willReturn(3);
 
-        // when & then
         assertThrows(GoalSelectionLimitExceededException.class,
                 () -> learningGoalService.selectGoal(userId, goalMasterId));
 
+        verify(userRepository).findById(userId);
         verify(goalMasterRepository).findById(goalMasterId);
-        verify(userGoalRepository).existsByUserIdAndGoalMaster_GoalMasterId(userId, goalMasterId);
-        verify(userGoalRepository).countByUserId(userId);
+        verify(userGoalRepository).existsByUser_UserIdAndGoalMaster_GoalMasterId(userId, goalMasterId);
+        verify(userGoalRepository).countByUser_UserId(userId);
         verify(userGoalRepository, never()).save(any(UserGoal.class));
     }
 
     @Test
     @DisplayName("진행 중인 목표를 정상적으로 해제한다")
     void 진행_중인_목표를_정상적으로_해제한다() {
-        // given
         Long userGoalId = 10L;
+
         UserGoal userGoal = UserGoal.of(
                 userGoalId,
-                userId,
+                user,
                 goalMaster,
                 0,
                 UserGoalStatus.ACTIVE,
-                java.time.LocalDate.now(),
-                java.time.LocalDate.now()
+                LocalDate.now(),
+                LocalDate.now()
         );
 
         given(userGoalRepository.findById(userGoalId))
                 .willReturn(Optional.of(userGoal));
 
-        // when
         learningGoalService.cancelGoal(userGoalId);
 
-        // then
         verify(userGoalRepository).findById(userGoalId);
         assertEquals(UserGoalStatus.CANCELED, userGoal.getStatus());
     }
@@ -185,13 +211,11 @@ class LearningGoalServiceTest {
     @Test
     @DisplayName("존재하지 않는 사용자 목표는 해제할 수 없다")
     void 존재하지_않는_사용자_목표는_해제할_수_없다() {
-        // given
         Long userGoalId = 999L;
 
         given(userGoalRepository.findById(userGoalId))
                 .willReturn(Optional.empty());
 
-        // when & then
         assertThrows(UserGoalNotFoundException.class,
                 () -> learningGoalService.cancelGoal(userGoalId));
 
@@ -201,10 +225,9 @@ class LearningGoalServiceTest {
     @Test
     @DisplayName("사용자의 진행 중인 목표 목록을 조회한다")
     void 사용자의_진행_중인_목표_목록을_조회한다() {
-        // given
         UserGoal activeGoal1 = UserGoal.of(
                 1L,
-                userId,
+                user,
                 goalMaster,
                 0,
                 UserGoalStatus.ACTIVE,
@@ -214,7 +237,7 @@ class LearningGoalServiceTest {
 
         UserGoal activeGoal2 = UserGoal.of(
                 2L,
-                userId,
+                user,
                 goalMaster,
                 0,
                 UserGoalStatus.ACTIVE,
@@ -222,14 +245,12 @@ class LearningGoalServiceTest {
                 LocalDate.now()
         );
 
-        given(userGoalRepository.findAllByUserIdAndStatus(userId, UserGoalStatus.ACTIVE))
-                .willReturn(java.util.List.of(activeGoal1, activeGoal2));
+        given(userGoalRepository.findAllByUser_UserIdAndStatus(userId, UserGoalStatus.ACTIVE))
+                .willReturn(List.of(activeGoal1, activeGoal2));
 
-        // when
-        java.util.List<UserGoal> result = learningGoalService.getActiveGoals(userId);
+        List<UserGoal> result = learningGoalService.getActiveGoals(userId);
 
-        // then
-        verify(userGoalRepository).findAllByUserIdAndStatus(userId, UserGoalStatus.ACTIVE);
+        verify(userGoalRepository).findAllByUser_UserIdAndStatus(userId, UserGoalStatus.ACTIVE);
         assertEquals(2, result.size());
         assertEquals(UserGoalStatus.ACTIVE, result.get(0).getStatus());
         assertEquals(UserGoalStatus.ACTIVE, result.get(1).getStatus());
@@ -238,7 +259,6 @@ class LearningGoalServiceTest {
     @Test
     @DisplayName("진행 중인 목표의 진행도를 정상적으로 증가시킨다")
     void 진행중인_목표의_진행도를_정상적으로_증가시킨다() {
-
         GoalMaster progressGoalMaster = GoalMaster.of(
                 goalMasterId,
                 "하루 3회 학습",
@@ -249,7 +269,7 @@ class LearningGoalServiceTest {
 
         UserGoal userGoal = UserGoal.of(
                 1L,
-                userId,
+                user,
                 progressGoalMaster,
                 1,
                 UserGoalStatus.ACTIVE,
@@ -257,12 +277,12 @@ class LearningGoalServiceTest {
                 LocalDate.now()
         );
 
-        given(userGoalRepository.findAllByUserIdAndStatus(userId, UserGoalStatus.ACTIVE))
+        given(userGoalRepository.findAllByUser_UserIdAndStatus(userId, UserGoalStatus.ACTIVE))
                 .willReturn(List.of(userGoal));
 
         learningGoalService.updateGoalProgress(userId, GoalType.STUDY_COUNT, 1);
 
-        verify(userGoalRepository).findAllByUserIdAndStatus(userId, UserGoalStatus.ACTIVE);
+        verify(userGoalRepository).findAllByUser_UserIdAndStatus(userId, UserGoalStatus.ACTIVE);
         assertEquals(2, userGoal.getCurrentValue());
         assertEquals(UserGoalStatus.ACTIVE, userGoal.getStatus());
     }
@@ -270,7 +290,6 @@ class LearningGoalServiceTest {
     @Test
     @DisplayName("목표 진행도가 목표값에 도달하면 완료 처리한다")
     void 목표_진행도가_목표값에_도달하면_완료_처리한다() {
-
         GoalMaster progressGoalMaster = GoalMaster.of(
                 goalMasterId,
                 "하루 3회 학습",
@@ -281,7 +300,7 @@ class LearningGoalServiceTest {
 
         UserGoal userGoal = UserGoal.of(
                 1L,
-                userId,
+                user,
                 progressGoalMaster,
                 2,
                 UserGoalStatus.ACTIVE,
@@ -289,12 +308,12 @@ class LearningGoalServiceTest {
                 LocalDate.now()
         );
 
-        given(userGoalRepository.findAllByUserIdAndStatus(userId, UserGoalStatus.ACTIVE))
+        given(userGoalRepository.findAllByUser_UserIdAndStatus(userId, UserGoalStatus.ACTIVE))
                 .willReturn(List.of(userGoal));
 
         learningGoalService.updateGoalProgress(userId, GoalType.STUDY_COUNT, 1);
 
-        verify(userGoalRepository).findAllByUserIdAndStatus(userId, UserGoalStatus.ACTIVE);
+        verify(userGoalRepository).findAllByUser_UserIdAndStatus(userId, UserGoalStatus.ACTIVE);
         assertEquals(3, userGoal.getCurrentValue());
         assertEquals(UserGoalStatus.COMPLETED, userGoal.getStatus());
     }
@@ -302,10 +321,9 @@ class LearningGoalServiceTest {
     @Test
     @DisplayName("사용자의 완료된 목표 목록을 조회한다")
     void 사용자의_완료된_목표_목록을_조회한다() {
-
         UserGoal completedGoal1 = UserGoal.of(
                 1L,
-                userId,
+                user,
                 goalMaster,
                 1,
                 UserGoalStatus.COMPLETED,
@@ -314,8 +332,8 @@ class LearningGoalServiceTest {
         );
 
         UserGoal completedGoal2 = UserGoal.of(
-                1L,
-                userId,
+                2L,
+                user,
                 goalMaster,
                 1,
                 UserGoalStatus.COMPLETED,
@@ -323,12 +341,12 @@ class LearningGoalServiceTest {
                 LocalDate.now()
         );
 
-        given(userGoalRepository.findAllByUserIdAndStatus(userId, UserGoalStatus.COMPLETED))
+        given(userGoalRepository.findAllByUser_UserIdAndStatus(userId, UserGoalStatus.COMPLETED))
                 .willReturn(List.of(completedGoal1, completedGoal2));
 
         List<UserGoal> result = learningGoalService.getCompletedGoals(userId);
 
-        verify(userGoalRepository).findAllByUserIdAndStatus(userId, UserGoalStatus.COMPLETED);
+        verify(userGoalRepository).findAllByUser_UserIdAndStatus(userId, UserGoalStatus.COMPLETED);
         assertEquals(2, result.size());
         assertEquals(UserGoalStatus.COMPLETED, result.get(0).getStatus());
         assertEquals(UserGoalStatus.COMPLETED, result.get(1).getStatus());
@@ -337,14 +355,13 @@ class LearningGoalServiceTest {
     @Test
     @DisplayName("사용자가 아직 선택하지 않은 목표 목록을 조회한다")
     void 사용자가_아직_선택하지_않은_목표_목록을_조회한다() {
-
         GoalMaster goal1 = GoalMaster.of(1L, "하루 1회 학습", GoalPeriodType.DAILY, GoalType.STUDY_COUNT, 1);
         GoalMaster goal2 = GoalMaster.of(2L, "하루 3회 학습", GoalPeriodType.DAILY, GoalType.STUDY_COUNT, 3);
         GoalMaster goal3 = GoalMaster.of(3L, "주간 말하기 5회", GoalPeriodType.WEEKLY, GoalType.SPEAKING_COUNT, 5);
 
         UserGoal selectedGoal = UserGoal.of(
                 10L,
-                userId,
+                user,
                 goal1,
                 0,
                 UserGoalStatus.ACTIVE,
@@ -355,17 +372,16 @@ class LearningGoalServiceTest {
         given(goalMasterRepository.findAllByIsActiveTrue())
                 .willReturn(List.of(goal1, goal2, goal3));
 
-        given(userGoalRepository.findAllByUserIdAndStatus(userId, UserGoalStatus.ACTIVE))
+        given(userGoalRepository.findAllByUser_UserIdAndStatus(userId, UserGoalStatus.ACTIVE))
                 .willReturn(List.of(selectedGoal));
 
         List<GoalMaster> result = learningGoalService.getAvailableGoals(userId);
 
         verify(goalMasterRepository).findAllByIsActiveTrue();
-        verify(userGoalRepository).findAllByUserIdAndStatus(userId, UserGoalStatus.ACTIVE);
+        verify(userGoalRepository).findAllByUser_UserIdAndStatus(userId, UserGoalStatus.ACTIVE);
 
         assertEquals(2, result.size());
         assertEquals(2L, result.get(0).getGoalMasterId());
         assertEquals(3L, result.get(1).getGoalMasterId());
     }
-
 }

--- a/src/test/java/kr/co/mapspring/learning/service/LearningGoalServiceTest.java
+++ b/src/test/java/kr/co/mapspring/learning/service/LearningGoalServiceTest.java
@@ -92,18 +92,18 @@ class LearningGoalServiceTest {
         given(goalMasterRepository.findById(goalMasterId))
                 .willReturn(Optional.of(goalMaster));
 
-        given(userGoalRepository.existsByUser_UserIdAndGoalMaster_GoalMasterId(userId, goalMasterId))
+        given(userGoalRepository.existsByUser_UserIdAndGoalMaster_GoalMasterIdAndStatus(userId, goalMasterId, UserGoalStatus.ACTIVE))
                 .willReturn(false);
 
-        given(userGoalRepository.countByUser_UserId(userId))
+        given(userGoalRepository.countByUser_UserIdAndStatus(userId, UserGoalStatus.ACTIVE))
                 .willReturn(2);
 
         learningGoalService.selectGoal(userId, goalMasterId);
 
         verify(userRepository).findById(userId);
         verify(goalMasterRepository).findById(goalMasterId);
-        verify(userGoalRepository).existsByUser_UserIdAndGoalMaster_GoalMasterId(userId, goalMasterId);
-        verify(userGoalRepository).countByUser_UserId(userId);
+        verify(userGoalRepository).existsByUser_UserIdAndGoalMaster_GoalMasterIdAndStatus(userId, goalMasterId, UserGoalStatus.ACTIVE);
+        verify(userGoalRepository).countByUser_UserIdAndStatus(userId, UserGoalStatus.ACTIVE);
 
         ArgumentCaptor<UserGoal> captor = ArgumentCaptor.forClass(UserGoal.class);
         verify(userGoalRepository).save(captor.capture());
@@ -132,8 +132,8 @@ class LearningGoalServiceTest {
 
         verify(userRepository).findById(userId);
         verify(goalMasterRepository).findById(goalMasterId);
-        verify(userGoalRepository, never()).existsByUser_UserIdAndGoalMaster_GoalMasterId(any(), any());
-        verify(userGoalRepository, never()).countByUser_UserId(any());
+        verify(userGoalRepository, never()).existsByUser_UserIdAndGoalMaster_GoalMasterIdAndStatus(any(), any(), any());
+        verify(userGoalRepository, never()).countByUser_UserIdAndStatus(any(), any());
         verify(userGoalRepository, never()).save(any(UserGoal.class));
     }
 
@@ -146,7 +146,7 @@ class LearningGoalServiceTest {
         given(goalMasterRepository.findById(goalMasterId))
                 .willReturn(Optional.of(goalMaster));
 
-        given(userGoalRepository.existsByUser_UserIdAndGoalMaster_GoalMasterId(userId, goalMasterId))
+        given(userGoalRepository.existsByUser_UserIdAndGoalMaster_GoalMasterIdAndStatus(userId, goalMasterId, UserGoalStatus.ACTIVE))
                 .willReturn(true);
 
         assertThrows(GoalAlreadySelectedException.class,
@@ -154,8 +154,8 @@ class LearningGoalServiceTest {
 
         verify(userRepository).findById(userId);
         verify(goalMasterRepository).findById(goalMasterId);
-        verify(userGoalRepository).existsByUser_UserIdAndGoalMaster_GoalMasterId(userId, goalMasterId);
-        verify(userGoalRepository, never()).countByUser_UserId(any());
+        verify(userGoalRepository).existsByUser_UserIdAndGoalMaster_GoalMasterIdAndStatus(userId, goalMasterId, UserGoalStatus.ACTIVE);
+        verify(userGoalRepository, never()).countByUser_UserIdAndStatus(any(), any());
         verify(userGoalRepository, never()).save(any(UserGoal.class));
     }
 
@@ -168,10 +168,10 @@ class LearningGoalServiceTest {
         given(goalMasterRepository.findById(goalMasterId))
                 .willReturn(Optional.of(goalMaster));
 
-        given(userGoalRepository.existsByUser_UserIdAndGoalMaster_GoalMasterId(userId, goalMasterId))
+        given(userGoalRepository.existsByUser_UserIdAndGoalMaster_GoalMasterIdAndStatus(userId, goalMasterId, UserGoalStatus.ACTIVE))
                 .willReturn(false);
 
-        given(userGoalRepository.countByUser_UserId(userId))
+        given(userGoalRepository.countByUser_UserIdAndStatus(userId, UserGoalStatus.ACTIVE))
                 .willReturn(3);
 
         assertThrows(GoalSelectionLimitExceededException.class,
@@ -179,8 +179,8 @@ class LearningGoalServiceTest {
 
         verify(userRepository).findById(userId);
         verify(goalMasterRepository).findById(goalMasterId);
-        verify(userGoalRepository).existsByUser_UserIdAndGoalMaster_GoalMasterId(userId, goalMasterId);
-        verify(userGoalRepository).countByUser_UserId(userId);
+        verify(userGoalRepository).existsByUser_UserIdAndGoalMaster_GoalMasterIdAndStatus(userId, goalMasterId, UserGoalStatus.ACTIVE);
+        verify(userGoalRepository).countByUser_UserIdAndStatus(userId, UserGoalStatus.ACTIVE);
         verify(userGoalRepository, never()).save(any(UserGoal.class));
     }
 

--- a/src/test/java/kr/co/mapspring/learning/service/LearningServiceTest.java
+++ b/src/test/java/kr/co/mapspring/learning/service/LearningServiceTest.java
@@ -7,6 +7,12 @@ import kr.co.mapspring.learning.enums.StudyType;
 import kr.co.mapspring.learning.repository.StudyLogRepository;
 import kr.co.mapspring.learning.repository.StudyScoreRepository;
 import kr.co.mapspring.learning.service.impl.LearningServiceImpl;
+import kr.co.mapspring.place.entity.LearningSession;
+import kr.co.mapspring.place.repository.LearningSessionRepository;
+import kr.co.mapspring.user.entity.User;
+import kr.co.mapspring.user.enums.UserRole;
+import kr.co.mapspring.user.enums.UserStatus;
+import kr.co.mapspring.user.repository.UserRepository;
 import org.junit.jupiter.api.DisplayName;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.ExtendWith;
@@ -14,10 +20,16 @@ import org.mockito.ArgumentCaptor;
 import org.mockito.InjectMocks;
 import org.mockito.Mock;
 import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.test.util.ReflectionTestUtils;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.util.Optional;
 
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.mockito.ArgumentMatchers.any;
 import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.verify;
 
 @ExtendWith(MockitoExtension.class)
@@ -32,13 +44,18 @@ class LearningServiceTest {
     @Mock
     private LearningGoalService learningGoalService;
 
+    @Mock
+    private UserRepository userRepository;
+
+    @Mock
+    private LearningSessionRepository learningSessionRepository;
+
     @InjectMocks
     private LearningServiceImpl learningService;
 
     @Test
     @DisplayName("학습 기록과 점수를 정상적으로 저장한다")
     void 학습_기록과_점수를_정상적으로_저장한다() {
-
         Long userId = 1L;
         Long sessionId = 10L;
         StudyType studyType = StudyType.SCENARIO;
@@ -47,6 +64,30 @@ class LearningServiceTest {
         Integer naturalnessScore = 80;
         Integer fluencyScore = 70;
         Integer totalScore = 75;
+
+        User user = User.builder()
+                .userId(userId)
+                .email("test@test.com")
+                .name("테스트")
+                .birthDate(LocalDate.of(2000, 1, 1))
+                .address("서울")
+                .phoneNumber("010-1234-5678")
+                .passwordHash("1234")
+                .status(UserStatus.ACTIVE)
+                .role(UserRole.USER)
+                .build();
+
+        LearningSession session = mock(LearningSession.class);
+
+        ReflectionTestUtils.setField(session, "sessionId", sessionId);
+        ReflectionTestUtils.setField(session, "startTime", LocalDateTime.now());
+        ReflectionTestUtils.setField(session, "user", user);
+
+        given(userRepository.findById(userId))
+                .willReturn(Optional.of(user));
+
+        given(learningSessionRepository.findById(sessionId))
+                .willReturn(Optional.of(session));
 
         given(studyLogRepository.save(any(StudyLog.class)))
                 .willAnswer(invocation -> invocation.getArgument(0));
@@ -64,14 +105,16 @@ class LearningServiceTest {
         ArgumentCaptor<StudyLog> studyLogCaptor = ArgumentCaptor.forClass(StudyLog.class);
         ArgumentCaptor<StudyScore> studyScoreCaptor = ArgumentCaptor.forClass(StudyScore.class);
 
+        verify(userRepository).findById(userId);
+        verify(learningSessionRepository).findById(sessionId);
         verify(studyLogRepository).save(studyLogCaptor.capture());
         verify(studyScoreRepository).save(studyScoreCaptor.capture());
 
         StudyLog savedStudyLog = studyLogCaptor.getValue();
         StudyScore savedStudyScore = studyScoreCaptor.getValue();
 
-        assertEquals(userId, savedStudyLog.getUserId());
-        assertEquals(sessionId, savedStudyLog.getSessionId());
+        assertEquals(user, savedStudyLog.getUser());
+        assertEquals(session, savedStudyLog.getLearningSession());
         assertEquals(studyType, savedStudyLog.getStudyType());
         assertEquals(earnedExp, savedStudyLog.getEarnedExp());
 
@@ -82,5 +125,4 @@ class LearningServiceTest {
 
         verify(learningGoalService).updateGoalProgress(userId, GoalType.STUDY_COUNT, 1);
     }
-
 }

